sound/soc/sof/pm.c

```diff
@@ -305,15 +305,19 @@ EXPORT_SYMBOL(snd_sof_suspend);
 int snd_sof_prepare(struct device *dev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
+	const struct sof_dev_desc *desc = sdev->pdata->desc;
+
+	/* will suspend to S3 by default */
+	sdev->system_suspend_target = SOF_SUSPEND_S3;
+
+	if (!desc->acpi_capable)
+		return 0;
 
 #if defined(CONFIG_ACPI)
 	if (acpi_target_system_state() == ACPI_STATE_S0)
 		sdev->system_suspend_target = SOF_SUSPEND_S0IX;
 	else
 		sdev->system_suspend_target = SOF_SUSPEND_S3;
```

Comment:
@dbaluta this else is no longer needed isnt it?

Reply:
@ranj063 yes, I've seen this. But at least in the first version of the patch I wanted to keep it in order to have fewer changes. 

Also, care to explain a little bit what is acpi_target_system_state and what is system_suspend_target. Also, one can notice that if CONFIG_ACPI is not defined acpi_target_system_state by default is ACPI_STATE_S3.